### ISSUE TITLE
Questions about the update of memory bank

### ISSUE THREAD

**Issue by Liuyang829** [maple-research-lab/AdCo#4]
Thanks for your great work! I have several questions about this work. The paper doesn't explain how to update the negative adversaries and I am wonder about some details of the update. I suppose that the normal `loss.backward()` in your code is just the same as InfoNCE loss like normal contrastive learning, and the update of negative adversaries is up to manual update of memory bank, I want to know some details about this. 
Why we need logits in the update and what dose the mem_losses mean; this loss doesn't seem to be able to update automatically by pytorch. How does maximize the contrastive loss to update negative adversaries reflect in your code
Thank you very much.

**Comment by wang3702** [comment]
Simple conclusion is that the update of the memory bank is just -contrastive_loss with another temperature. Our implementation is based on the gradient calculation as shown in our paper. This complicated implementation can accelerate the training time. If you are still confused, please follow our section 3.3 fully addressed this implementation.
